content/blog/2025-05-08-go-client-GA.md

```diff
@@ -0,0 +1,501 @@
+
+## Standalone Mode Batch
+
+```go
+    // Standalone BatchClient Example
+	cfg := config.NewClientConfiguration().
+		WithAddress(&config.NodeAddress{Host: "localhost", Port: 6379})
+
+	client, err := NewClient(cfg)
+	if err != nil {
+		log.Fatal("error connecting to database: ", err)
+	}
+	defer client.Close()
+
+	// Create new transaction
+	tx := pipeline.NewStandaloneBatch(true).
+		Set("user:profile", "John Doe").
+		Set("user:email", "john@example.com").
+		SAdd("users", []string{"john"})
+
+	// Command sequence
+	txs := []string{"Set - user:profile", "Set - user:email", "SAdd - users"}
+
+	// Execute transaction
+	results, err := client.Exec(context.Background(), *tx, false) // false = process all commands and include errors as part of the response array
+	if err != nil {
+		log.Printf("Transaction failed: %v", err)
+		return
+	}
+
+	// Process results
+	for i, result := range results {
+		fmt.Printf("Command %s results: %v\n", txs[i], result)
+	}
+```
+
+## Cluster mode Batch
+
+Cluster batches are similar to standalone batches, but they are executed on a cluster client. They may have specific routing options to control which node the commands are executed on. In a cluster batch, all commands are executed on the same node, unless the IsAtomic flag is set to false (i.e. is a pipeline and not a transaction), in which case the commands may be split based on the key slot. Once all commands have been executed, the results (including any errors) are returned to the client. For pipelines, the results are returned in the same order as the commands were queued, even when the commands are executed on different nodes.
+
+```go
+	// Cluster BatchClient Example
+    // Only a single node is required to connect to the Cluster
+	cfg := config.NewClusterClientConfiguration().
+		WithAddress(&config.NodeAddress{Host: "localhost", Port: 7010})
+
+	clusterClient, err := NewClusterClient(cfg)
+	if err != nil {
+		log.Fatal("error connecting to cluster: ", err)
+	}
+	defer clusterClient.Close()
+
+	// Create new transaction
+	clusterTx := pipeline.NewClusterBatch(true)
+
+	// Create ping options with specific routing
+	pingOpts := options.PingOptions{
+		Message: "Hello Valkey Cluster",
+	}
+
+	// Queue commands with routing
+	clusterTx.PingWithOptions(pingOpts)
+
+	// Command sequence
+	txs := []string{"Ping - 'Hello Valkey Cluster'"}
+
+	opts := pipeline.NewClusterBatchOptions().
+		WithRoute(config.RandomRoute)
+
+	// Execute transaction
+	results, err := clusterClient.ExecWithOptions(context.Background(), *clusterTx, false, *opts)
+	if err != nil {
+		log.Printf("Cluster transaction failed: %v", err)
+		return
+	}
+
+	// Process results
+	for i, result := range results {
+		fmt.Printf("Command %s results: %v\n", txs[i], result)
+	}
+```
```

Comment:
Maybe show an example of receiving error with `raiseOnError` set and unset (see `TestBatchRaiseOnError`)

Reply:
Blog post is already quite long. I don't think we have room to cover every single option and feature. We have documentation for that.